Firestore/Source/Local/FSTMemoryPersistence.mm

```diff
@@ -48,19 +59,35 @@ @implementation FSTMemoryPersistence {
   /** The FSTRemoteDocumentCache representing the persisted cache of remote documents. */
   FSTMemoryRemoteDocumentCache *_remoteDocumentCache;
 
-  std::unordered_map<User, id<FSTMutationQueue>, HashUser> _mutationQueues;
-
   FSTTransactionRunner _transactionRunner;
+
+  id<FSTReferenceDelegate> _referenceDelegate;
+}
+
++ (instancetype)persistenceWithEagerGC {
+  return [[FSTMemoryPersistence alloc]
```

Comment:
It's much more conventional to hook up delegates just as a separate statement, something like:

```objc
FSTMemoryEagerReferenceDelegate *referenceDelegate = [[... alloc] init];
FSTMemoryPersistence *persistence = [[FSTMemoryPersistence alloc] initWithReferenceDelegate:referenceDelegate];
referenceDelegate.persistence = persistence;
```

Or the other way around.

It's important to establish which is a delegate of the other because as-is you've created a reference cycle, which will prevent either object from collected by arc.

Since the persistence object is retained by the local store (and not the reference delegate), likely the reference delegate should have a weak reference back to the persistence object and the initialization sequence above is the right one.

Reply:
Assigned delegates procedurally and marked the delegate's reference to `persistence` as `__weak` to avoid a retain cycle. Left a comment noting that the delegate should live as long as the underlying persistence layer.